packages/api-codec/src/Metadata.ts

```diff
@@ -194,7 +191,14 @@ class StorageFunctionType extends CodecEnumType<Type | StorageFunctionType$Map>
   }
 }
 
-class StorageFunctionMetadata extends CodecStruct {
+export interface IStorageFunctionMetadata {
```

Comment:
Really not satisfied with this, looks too java-like. But I needed StorageFunctionMetadata in type-storage, without `byteLenght` and `from*`

Reply:
Left a suggestion for you there...